File: 287_find_duplicate_num/sort.py
class Solution:
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        """
        Complexity: O(nlgn) quick sort avg case
        Space: O(1)
        """
        if nums == []:
            return -1
        sort_nums = sorted(nums)
        last_element = 0
        for ele in sort_nums:
            if ele == last_element:
                return ele
            else:
                last_element = ele

File: 581_Shortest_Unsorted_Continuous_Subarray/sort.cpp
class Solution {
  /*
   * Complexity: O(nlgn)
   * Space: O(n)
   */
public:
    int findUnsortedSubarray(vector<int>& nums) {
        vector<int>copy(nums.begin(), nums.end());
        sort(copy.begin(), copy.end());
        int n = nums.size();
        int start = -1, end = -1;
        
        for(int i = 0; i < n; ++i){
            if(copy[i] != nums[i]){
                if(start < 0) start = i;
                end = i;
            }
        }
        
        return (start < 0) ? 0 : (end - start + 1);
    }
};
